### PR TITLE
Add test grid scenario for Artemis connector

### PR DESCRIPTION
**Pull request by riyafa**
https://docs.wso2.com/display/ESB490/JMS+Synchronous+Invocations+%3A+Dual+Channel+HTTP-to-JMS

## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > This is a scenario test as described above

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

    - OracleJDK-8

### CODE REVIEW

File: connectors/artemis/src/test/java/org/ballerinalang/scenario/test/connector/artemis/ArtemisTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @Test(description = "Test Artemis connector")
+    public void testArtemisConnector() throws IOException {
+        String url = "http://" + host + ":" + port + "/artemis/test";
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        HttpResponse httpResponse = HttpClientRequest.doPost(url, "{\"Hi\":\"World\"}", headers);
+        Assert.assertEquals(httpResponse.getResponseCode(), 200, "Response code mismatching");
+        Assert.assertEquals(httpResponse.getData(), "{\"hello\":\"Riyafa\"}");
+    }
+}
```
Comment (Manuri) [inline connectors/artemis/src/test/java/org/ballerinalang/scenario/test/connector/artemis/ArtemisTest.java]: Is this the only scenario we need to test? Aren't there negative cases etc?

File: pom.xml [hunk 2]
```diff
@@ -115,6 +116,13 @@
                 <module>connectors/soap</module>
             </modules>
         </profile>
+        <profile>
+            <id>artemis</id>
```
Comment (Manuri) [inline pom.xml]: let's name this messaging-artemis.

In the beginning I thought of having separate profiles for separate areas eg: http, database, messaging and have packages or modules inside that profile and filter out relevant module when the tests are running. This is just to avoid the pom being cluttered with lots of profiles.
When doing that we need to make sure, when running tests for a given messaging connector, other connectors aren't getting unnecessarily built.

Anyway, for now let's keep this way and address it when adding a new messaging connector.

Comment (riyafa) [reply pom.xml]: Done

File: connectors/artemis/src/test/java/org/ballerinalang/scenario/test/connector/artemis/ArtemisTest.java (hunk 1 above)
Comment (riyafa) [reply connectors/artemis/src/test/java/org/ballerinalang/scenario/test/connector/artemis/ArtemisTest.java]: Yes it is the only scenario test in this PR. We hope to add more tests on top of this framework.
